Add dladm_iptun module

xen0l opened the pull request:
##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - New Module Pull Request


##### COMPONENT NAME
dladm_iptun

##### ANSIBLE VERSION
latest

##### SUMMARY
Module for managing IP tunnel interfaces on illumos/Solaris.

xen0l committed:
Add dladm_iptun module

privateip commented on lib/ansible/modules/network/illumos/dladm_iptun.py:
```diff
@@ -0,0 +1,247 @@
+
+    module.exit_json(**result)
+
+from ansible.module_utils.basic import *
```
please use explicit imports `from ansible.module_utils import AnsibleModule`

privateip commented on lib/ansible/modules/network/illumos/dladm_iptun.py:
```diff
@@ -0,0 +1,247 @@
+    module.exit_json(**result)
+
+from ansible.module_utils.basic import *
+main()
```
please wrap in block

```
if __name__ == '__main__':
   main()
```

privateip reviewed (changes requested):
* missing RETURN doc string
* missing METADATA 
* please consider adding unit tests

xen0l committed:
Add ANSIBLE_METADATA
Document return values
Make imports explicit
Use native YAML syntax in EXAMPLES

xen0l commented:
@privateip I addressed you comments. As for unit tests, they will be shared among all modules to a certain degree (I am also speaking about modules, which are already included with Ansible). For that reason, I would like to add them in the next round if possible.

xen0l commented:
ready_for_review
